src/lib/forward.ts

```diff
@@ -103,9 +111,10 @@ export class EventForwarder {
     {
       const events: ZWaveNodeEvents[] = ["interview completed", "ready"];
       for (const event of events) {
-        node.on(event, (changedNode: ZWaveNode) =>
-          notifyNode(changedNode, event)
-        );
+        node.on(event, (changedNode: ZWaveNode) => {
+          const nodeState = dumpNode(changedNode);
```

Comment:
The description mention doing this only for `ready` but here you're doing the same for `interview completed`. Is that expected?

Reply:
Well, I did that just in case to be complete. So for both ready and interview completed the full node is dumped.
"interview completed" fires after "ready"

Reply:
`interview completed` means that all value metadata are updated. So it makes sense

Reply:
Just keep in mind that, for battery devices, the ready event could take so long before its emitted, even more for the interview completed

Reply:
We might even want to process "interview completed" in the lib to do our magic instead of ready

Reply:
> Just keep in mind that, for battery devices, the ready event could take so long before its emitted, even more for the interview completed

Yeah, that's why we send the basic node info before that so we can at least show the node info that is already present and create HA entities after all metadata is there.

Reply:
@AlCalzone could make better suggestions then me here

Reply:
I've been watching the events just now and "ready" is emitted before "interview completed" and some values get added in between the two. So I guess we should consider "interview completed" as our "ready to do something" rather than "node ready" ?

Reply:
Depends on the use case. After the very first interview, `ready` and `interview completed` are emitted back to back
During the following interviews, `ready` is emitted very soon, since we have all information about the node to interact with it and all metadata for all values. `interview completed` is then raised when all values themselves are updated, but that might take very long.
I advise listening for `ready` and treating everything after that as updated values.

Reply:
Thanks! So when "ready" is emitted all metadata etc. will be there. No need to wait for "interview completed" ?

Reply:
nope, only if you absolutely need up to date values

Reply:
Agreed. This is already handled by the fact that value events are emitted after the node is ready.
All clear now then.